### PR TITLE
initial version, extracted from leadconduit-integration 0.3.0

### PR DESCRIPTION
**Pull request by cgrayson**
The single test is admittedly pretty thin, but I wanted `cake test` to at least not fail with "no spec/ found".

### CODE REVIEW

**Comment by cgrayson** [comment]
Sorry, hang on :-/

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+lib
+node_modules
+*.tgz
```
Comment (alexkwolfe) [inline .gitignore]: I don't think you need this.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+# Common LeadConduit Integration Cakefile
+
+A Node.JS utility module for building LeadConduit integrations.
+
+[![Build Status](https://travis-ci.org/activeprospect/leadconduit-cakefile.svg?branch=master)](https://travis-ci.org/activeprospect/leadconduit-cakefile)
+
+
+This module provides the standard build utilities for integration modules. To use it:
+
+1. require in your module's `package.json` (of course)
```
Comment (alexkwolfe) [inline README.md]: Maybe this could be `npm install leadconduit-cakefile --save-dev`?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,29 @@
...
+2. existing modules may have `coffee-script` and `mocha` in their `DevDependencies` - those can (probably) be removed.
```
Comment (alexkwolfe) [inline README.md]: The first `D` in `DevDependencies` is lower case.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,29 @@
...
+
+The tasks provided for the `cake` command (run `cake` to see the list, as well):
+
+- `build` - delete and rebuild the package's `lib` directory from all `.coffee` source files in `src`
+- `lint` - run `coffeelint` on `.coffee` source files in `src`
+- `pdf` - generate versioned PDF files of all `.md` Markdown files in `docs`
+    - These are placed in the `drive` directory, if there is one, or left in the `docs` directory, if not. If you create a symlink to a local Google Drive copy of the appropriate subfolder in the ["Integrations" folder](https://drive.google.com/drive/folders/0B36C7G4bQKsAUkl5NWFFazZXaHc), called `drive`, and you keep the Google Drive sync app running, this provides a kind of automatic uploading of doc PDFs for use by the company at large.
+- `test` - run the mocha unit tests defined in `*-spec.coffee` spec files in `spec`
+    - Optionally, use `-p` or `--path` to specify a prefix for the test files to run. This can be a string which prefixes the files to target (e.g., "inbound" would match just `inbound*-spec.coffee`), or with a trailing slash, a directory name (e.g., "outbound/" would match all the `*-spec.coffee` files in `spec/outbound`).
```
Comment (alexkwolfe) [inline README.md]: Should we call this option `--prefix`?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,29 @@
...
+
+This module provides the standard build utilities for integration modules. To use it:
+
+1. require in your module's `package.json` (of course)
+2. existing modules may have `coffee-script` and `mocha` in their `DevDependencies` - those can (probably) be removed.
+3. create a file called `Cakefile` in the root of the module, with this single line:
+
+```coffeescript
+require('leadconduit-integration').cake(task)
+```
```
Comment (alexkwolfe) [inline README.md]: I think maybe this require is wrong? Shouldn't it be `require('leadconduit-cakefile')`? Also, if you wanted, you could eliminate the `cake` function. Maybe you thought about it and discarded that idea.

```coffeescript
require('leadconduit-cakefile')(task)
```

**Comment by alexkwolfe** [comment]
Otherwise, L:cake:TM

**Comment by alexkwolfe** [comment]
Have you tried overloading one of the cake tasks defined by this module? For example, if an integration wanted to use this module, but then wanted to have a custom `build` task, could that be done?

File: README.md (hunk 5 above)
Comment (cgrayson) [inline README.md]: I'm honestly a little hand-wavy on how this works, but I can attest that it does work as I wrote it. I'm not sure how (or why) to eliminate the `cake` function, nor do I see how your first question omits the `(task)` part but your block example doesn't.

**Pull request merged**